.travis.yml

```diff
@@ -46,174 +43,256 @@ cache:
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, libgmp-dev]}}
 
   # Build on OS X in addition to Linux
+  - env: BUILD=stack ARGS=" "
+    compiler: ": #stack default osx"
+    os: osx
+
   - env: BUILD=stack ARGS="--resolver lts-2"
+    compiler: ": #stack 7.8.4 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-3"
+    compiler: ": #stack 7.10.2 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-5"
+    compiler: ": #stack 7.10.3 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver nightly"
+    compiler: ": #stack nightly osx"
     os: osx
 
   allow_failures:
-  - env: BUILD=cabal GHCVER=head  CABALVER=head
+  - env: BUILD=cabal GHCVER=head  CABALVER=head HAPPYVER=1.19.5 ALEXVER=3.1.7
   - env: BUILD=stack ARGS="--resolver nightly"
 
-before_install:
 
+install:
+# SYMENGINE INSTALL PHASE
+# -----------------------
+# Download and install SymEngine
+- |
+  set -ex
+  export our_install_dir=$HOME/symengine-install
+  mkdir -p $our_install_dir
+  cd $HOME && git clone https://github.com/symengine/symengine.git
+ 
+# Setup C compiler variables
+# The reason we need to do this is because our build system is haskell, so none of the 
+# C variables are set.
+- |
+  if [ `uname` = "Darwin" ]
+  then
+    export CC="clang" && export CXX="clang++" && export CMAKE_C_COMPILER=$CC && export CMAKE_CXX_COMPILER=$CXX
+
+    # build gmp with clang
+    wget https://raw.githubusercontent.com/symengine/dependencies/6a42d290071921a0a478c6883fc0ddd709d664c9/gmp-6.0.0a.tar.bz2
+    tar -xjf gmp-6.0.0a.tar.bz2;
+    cd gmp-6.0.0 && ./configure --prefix=$our_install_dir --enable-cxx && make -j8 install && cd ..;
+
+  else
+    export CC="gcc" && export CXX="g++-4.7" && export CMAKE_C_COMPILER=$CC && export CMAKE_CXX_COMPILER=$CXX  
+  fi
+
+- |
+  set -ex
+  export TEST_CPP="no"
+  cd $HOME/symengine
+  source bin/install_travis.sh
+  bin/test_travis.sh
+
+# EXPORT PHASE
+# ------------
+# Export environment variables related to SymEngine's library and includes
+- |
+   # $our_install_dir is exported by test_travis.sh from symengine
+   set -ex
+   export SYMENGINE_LIB_ARGS="--extra-lib-dirs=$our_install_dir/lib/"  
+   export SYMENGINE_INCLUDE_ARGS="--extra-include-dirs=$our_install_dir/include/ --extra-include-dirs=$our_install_dir/include/symengine/"
+   cd $TRAVIS_BUILD_DIR
+
+# GHC INSTALL PHASE
+# -----------------
+# Install Stack if needed, then install GHC
+
+# Using compiler above sets CC to an invalid value, so unset it
+- unset CC
 # We want to always allow newer versions of packages when building on GHC HEAD
 - CABALARGS=""
 - if [ "x$GHCVER" = "xhead" ]; then CABALARGS=--allow-newer; fi
 
 # Download and unpack the stack executable
-- export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$HOME/.local/bin:$PATH
+- export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$HOME/.local/bin:/opt/alex/$ALEXVER/bin:/opt/happy/$HAPPYVER/bin:$HOME/.cabal/bin:$PATH
```

Comment:
`Happy` is the parser generator, `Alex` is the lexer. They're both Haskell tools, and that's where the two get installed respectively as far as I know.